### ISSUE TITLE
Build error: NodeListOf is not an array type

### ISSUE THREAD

**Issue by willshowell** [angular/material.angular.io#238]
I get the following error when trying to serve the app:

```
ERROR in material.angular.io/src/app/shared/table-of-contents/table-of-contents.ts (105,28): Type 'NodeListOf<Element>' is not an array type or a string type.
```

This seems like a Typescript issue, but the package-lock shows 2.2.2. I've tried with npm@4.6.1, @5.0.3, and @5.3.0 to make sure it's not a package-lock issue installing an invalid version of Typescript. I've also tried with node versions 7.0.0 and 8.4.0.

Is there something I'm missing?

[As shown on SO](https://stackoverflow.com/a/45803577/4811678), the following diff gets rid of the error:
```diff
     if (headers.length) {
-      for (const header of headers) {
+      for (let i = 0; i < headers.length; i++) {
+        const header: any = headers[i];
         // remove the 'link' icon name from the inner text
         const name = header.innerText.replace(/^link/, '');
         const {top} = header.getBoundingClientRect();
```

**Comment by willshowell** [comment]
Note this is also causing [Saucelabs and BrowserStack failures](https://travis-ci.org/angular/material.angular.io/builds/265228967)

**Comment by josephperrott** [comment]
Fixed by #239
